ImageTransform/CommonPart/transform/Transformation.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "ColorCalc.hpp"
+#include "Constants.hpp"
+#include <QPoint>
+#include <QPointF>
+#include <QPolygon>
+#include <QPolygonF>
+#include <QRect>
+#include <QRectF>
+#include <cmath>
+#include <complex>
+#include <iostream>
+
+namespace common
+{
+
+template <typename T_dig, typename T_point, typename T_rect, typename T_poly>
+class Transform
+{
+
+public:
+    Transform(const T_point& rotCenter, T_dig deg_angle)
+        : m_rotCenter{rotCenter}, m_angle{deg_angle}
+    {
+    }
+
+    T_point operator()(const T_point& point)
+    {
+
+        T_point new_point = point - m_rotCenter;
+        std::complex<T_dig> new_point_cpl(new_point.x(), new_point.y());
+
+        T_dig angle = std::arg(new_point_cpl);
+
+        angle += new_point.y() < 0 ? constants::two_pi : 0;
+
+        T_dig module = std::abs(new_point_cpl);
+        m_angle += angle;
+
+        T_point out_point{module * cos(m_angle), module * sin(m_angle)};
+        out_point += m_rotCenter;
+        m_angle -= angle;
+        return out_point;
+    }
+
+    T_rect operator()(const T_rect& rect)
```

Comment:
This function should return `T_poly` because rect rotated by different angle than n*90° will be no longer possible to be described by topleft and bottomright points.

Reply:
You may convert rect to poly and redirect call to below operator